src/gui/accountsettings.cpp

```diff
@@ -671,6 +674,67 @@ void AccountSettings::slotAddFolder()
     folderWizard->open();
 }
 
+void AccountSettings::slotRenameFolder()
+{
+    FolderMan *folderMan = FolderMan::instance();
+    Folder *folder = FolderMan::instance()->folder(selectedFolderAlias());
+    QDir currentDirectory(folder->path());
+
+    auto dialog = new QDialog(this, Qt::WindowTitleHint | Qt::CustomizeWindowHint | Qt::WindowCloseButtonHint );
+    dialog->setSizeGripEnabled(true);
+    dialog->setSizePolicy(QSizePolicy(QSizePolicy::Expanding, QSizePolicy::Fixed));
+    dialog->setFixedHeight(100);
+    dialog->setMaximumWidth(this->width());
+    dialog->setWindowTitle("Rename local folder");
+
+    auto vbox = new QVBoxLayout;
+    auto lineEdit = new QLineEdit;
+
+    lineEdit->setText(currentDirectory.dirName());
+    vbox->addWidget(lineEdit);
+    auto hbox = new QHBoxLayout;
+    auto pushButtonConfirm = new QPushButton("Rename");
+    auto pushButtonCancel = new QPushButton("Cancel");
+    hbox->addWidget(pushButtonConfirm);
+    hbox->addWidget(pushButtonCancel);
+    vbox->addLayout(hbox);
+    dialog->setLayout(vbox);
+
+    connect(pushButtonConfirm, SIGNAL(clicked()), dialog, SLOT(accept()));
+    connect(pushButtonCancel, SIGNAL(clicked()), dialog, SLOT(reject()));
+
+    if (dialog->exec() == QDialog::Accepted) {
+        QString folderPath = currentDirectory.path().remove(currentDirectory.path().lastIndexOf(currentDirectory.dirName()), currentDirectory.dirName().length());
+        QString newFolderName = folderPath+lineEdit->text();
+
+        if (!(currentDirectory.path() == newFolderName)) {
```

Comment:
@tokun-mobica Please use a more simple condition `currentDirectory.path() != newFolderName`

Also, since `newFolderName` is a path, please rename it to `newFolderPath` just to avoid confusion when using it later with `QDir::rename(QDir::path(), ...)`